lib/puppet/agent.rb

```diff
@@ -95,7 +95,8 @@ def run_in_fork(forking = true)
         atForkHandler.child
         $0 = _("puppet agent: applying configuration")
         begin
-          exit(yield)
+          res = yield
+          res.nil? ? exit(1) : exit(res)
```

Comment:
Wouldn't `exit(yield || 1)` work here?

Reply:
It will also handle `false` value returned by block that is already handled in `exit`.
It should work, as ruby converts `false` to the same exit code `1`

Reply:
I didn't know that about `false -> 1`. Maybe better to be explicit using `yield || 1` so it's clear all falsey values return 1.

Reply:
done